appinventor/appengine/src/com/google/appinventor/client/actions/DeleteAction.java

```diff
@@ -0,0 +1,77 @@
+package com.google.appinventor.client.actions;
+
+import static com.google.appinventor.client.Ode.MESSAGES;
+
+import com.google.appinventor.client.ErrorReporter;
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.boxes.ProjectListBox;
+import com.google.appinventor.client.explorer.folder.ProjectFolder;
+import com.google.appinventor.client.explorer.project.Project;
+import com.google.gwt.user.client.Command;
+import com.google.gwt.user.client.Window;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeleteAction implements Command {
+  @Override
+  public void execute() {
+    Ode.getInstance().getEditorManager().saveDirtyEditors(new Command() {
+      @Override
+      public void execute() {
+        if (Ode.getInstance().getCurrentView() == Ode.PROJECTS) {
+          List<Project> selectedProjects =
+              ProjectListBox.getProjectListBox().getProjectList().getSelectedProjects();
+          List<ProjectFolder> selectedFolders = ProjectListBox.getProjectListBox().getProjectList().getSelectedFolders();
+          if (selectedProjects.size() > 0 || selectedFolders.size() > 0) {
+            List<Project> projectsToDelete = selectedProjects;
+            for(ProjectFolder f : selectedFolders) {
```

Comment:
I know this was just copied over from another file but code style should have a space between `for` and `(`